test/unit/go-binary-information-parser.spec.ts

```diff
@@ -0,0 +1,212 @@
+      "_cgo_gotypes.go",
+    ],
+  },
+  latest_normal: {
+    expectedFiles: [
+      "/usr/local/go/src/unicode/casetables.go",
+      "/usr/local/go/src/os/dir.go",
+      "/usr/local/go/src/io/io.go",
+      "/go/pkg/mod/github.com/ghodss/yaml@v1.0.0/yaml.go",
+      "/app/main.go",
+      "/go/pkg/mod/github.com/go-redis/redis/v9@v9.0.0-beta.2/redis.go",
+    ],
+  },
+  latest_stripped: {
+    expectedFiles: [
+      "/usr/local/go/src/unicode/casetables.go",
+      "/usr/local/go/src/io/io.go",
+      "/go/pkg/mod/github.com/ghodss/yaml@v1.0.0/yaml.go",
+      "/app/main.go",
+      "/go/pkg/mod/github.com/go-redis/redis/v9@v9.0.0-beta.2/redis.go",
+    ],
+  },
+  latest_cgo: {
+    expectedFiles: [
+      "/usr/local/go/src/unicode/casetables.go",
+      "/usr/local/go/src/io/io.go",
+      "/go/pkg/mod/github.com/ghodss/yaml@v1.0.0/yaml.go",
+      "/app/main.go",
+      "/go/pkg/mod/github.com/go-redis/redis/v9@v9.0.0-beta.2/redis.go",
+      "/usr/local/go/src/net/cgo_unix.go",
+      "/usr/local/go/src/runtime/cgo/asm_amd64.s",
+      "_cgo_gotypes.go",
+    ],
+  },
+  latest_cgo_trimmed_stripped: {
+    expectedFiles: [
+      "unicode/casetables.go",
+      "io/io.go",
+      "github.com/ghodss/yaml@v1.0.0/yaml.go",
+      "symboltest/main.go",
+      "github.com/go-redis/redis/v9@v9.0.0-beta.2/redis.go",
+      "net/cgo_unix.go",
+      "runtime/cgo/asm_amd64.s",
+      "_cgo_gotypes.go",
+    ],
+  },
+};
+
+describe("go symbols parser", () => {
+  const files = fs.readdirSync(pathJoin(__dirname, "../fixtures/go-binaries"), {
+    withFileTypes: true,
+  });
+  for (const file of files) {
+    if (!file.isFile()) {
+      continue;
+    }
+    const fileContent = fs.readFileSync(
+      pathJoin(__dirname, "../fixtures/go-binaries/", file.name),
+    );
+    const binary = elf.parse(fileContent) as any;
+    expect(isElfType(binary)).toBe(true);
+
+    it(`extract go version and modules from binary: ${file.name}`, async () => {
+      const res = extractModulesFromBinary(binary as Elf);
+      const expectedVersion = goVersionFromFileName(file.name);
+      if (expectedVersion !== "latest") {
+        expect(res.goVersion).toEqual(expectedVersion);
+      }
+      expect(res.modules).toMatchObject({
+        "github.com/gorilla/mux": "1.8.0",
+        "github.com/ghodss/yaml": "1.0.0",
+        "github.com/go-redis/redis/v9": "#beta.2",
+      });
+    });
+
+    if (testCases.hasOwnProperty(file.name)) {
+      const testCase = testCases[file.name];
+      it(`extract Go files from binary: ${file.name}`, async () => {
+        const pclnTab = binary.body.sections.find(
+          (section) => section.name === ".gopclntab",
+        );
+
+        const files = readFilesFromPCLNTable(pclnTab.data);
+        if (testCase.expectedFilesTotal !== undefined) {
+          expect(files).toHaveLength(testCase.expectedFilesTotal);
+        }
+        testCase.expectedFiles.forEach((file) => expect(files).toContain(file));
```

Comment:
you might want to read about `expect.arrayContaining` 
https://jestjs.io/docs/expect#expectarraycontainingarray

Reply:
Nice one, I wondered if there wasn't a better way to do this :smile: Thanks!